### PR TITLE
[release/6.0-rc1] Binding support for 'bool' values with InputRadioGroup and InputSelect

### PR DESCRIPTION
**Pull request by MackinnonBuck**
## Description

Added support for `bool` value bindings with the Blazor `InputRadioGroup` and `InputSelect` components.

## Customer Impact

Having radio/select inputs with only two choices is a fairly common scenario (true/false, yes/no prompts). In these cases, `bool` is an appropriate type for the value bound to the selected choice.

## Regression?
- [ ] Yes
- [X] No

## Risk
- [ ] High
- [ ] Medium
- [X] Low

This change is almost purely additive, so it's unlikely that it will break other functionality.

## Verification
- [X] Manual (required)
- [X] Automated

## Packaging changes reviewed?
- [ ] Yes
- [ ] No
- [X] N/A

Addresses #34816

### CODE REVIEW

**Comment by pranavkm** [comment]
FYI @davidfowl for approval.

**Review by pranavkm — APPROVED** [review]

**Comment by davidfowl** [comment]
Approved. I dunno if I should be using servicing approved 😄

**Review by davidfowl — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Looks like you didn't add a new API, but added an override which modified the unshipped file. Unfortunately this triggered the "don't modify APIs" check that we do for servicing branches.

We can probably modify https://github.com/dotnet/aspnetcore/blob/721c539f0f0aff1fbec77b7f4f6263ec5abec08d/eng/scripts/CodeCheck.ps1#L216 to allow `rc*` branches

File: eng/scripts/CodeCheck.ps1 [hunk 1]
```diff
@@ -213,7 +213,7 @@ try {
                     }
                 }
                 # Check for changes in Unshipped in servicing branches
-                if ($targetBranch -like 'release*' -and $targetBranch -notlike '*preview*' -and $file -like '*PublicAPI.Unshipped.txt') {
+                if ($targetBranch -like 'release*' -and $targetBranch -notlike '*preview*' -and $targetBranch -notlike '*rc*' -and $file -like '*PublicAPI.Unshipped.txt') {
```
Comment (BrennanConroy) [inline eng/scripts/CodeCheck.ps1:216]: cc @wtgodbe

Comment (wtgodbe) [reply eng/scripts/CodeCheck.ps1:216]: CC @dougbu as well

**Review by BrennanConroy — DISMISSED** [review]
I'm suspicious of this change, there are over 300 test failures in Components that look to have started around when this PR was being worked on.

You can see here on a Monday build of the PR https://dev.azure.com/dnceng/public/_build/results?buildId=1296055&view=results the tests were failing, and looking at any other test run of Components that don't have these changes there are nowhere near 300 test failures.

The reason you don't see any failures is because Steve accidentally switched the components pipeline to only run quarantined tests. You can see the 300 failures here https://github.com/dotnet/aspnetcore/pull/35484 where the tests are running again.

**Comment by MackinnonBuck** [comment]
Hm, this is strange. Even if the tests surrounding this change broke, I'm not sure why it would cause 300 seemingly unrelated tests to fail.

I'll investigate this as soon as I can to find out if there's something here causing the issues.

**Comment by MackinnonBuck** [comment]
Running on my machine locally, all tests except 1 pass.

**Comment by BrennanConroy** [comment]
Ok, after looking at more builds I think it's https://github.com/dotnet/aspnetcore/pull/35414 that caused the issues.

**Pull request merged**